lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -3346,9 +3346,7 @@ static void ReplaceMemcpy(Value *V, Value *Src, MemCpyInst *MC,
         const DataLayout &DL = SrcBCI->getModule()->getDataLayout();
         unsigned SrcSize = DL.getTypeAllocSize(
             SrcBCI->getOperand(0)->getType()->getPointerElementType());
-        unsigned MemcpySize =
-            cast<ConstantInt>(MC->getLength())->getZExtValue() *
-            MC->getAlignment();
+        unsigned MemcpySize = cast<ConstantInt>(MC->getLength())->getZExtValue();
```

Comment:
Could we make sure why MC->getAlignment() is not 1?

Reply:
Good idea, I put a conditional breakpoint, found it and fixed it. The code here is still wrong though so I'll keep my change here too.